drivers/adc/adc_ad559x.c

```diff
@@ -251,7 +251,7 @@ static int adc_ad559x_init(const struct device *dev)
 	return 0;
 }
 
-static const struct adc_driver_api adc_ad559x_api = {
+static DRIVER_API(adc_driver_api, adc_ad559x_api) = {
```

Comment:
note: we should be using the `DEVICE` namespace here, and now that we are at it, simplify things by obviating the `_driver_api` suffix (it's an impl detail required by syscalls, so it can be hidden).

Reply:
Replaced it with `DEVICE_` prefixes, I'm not keen on hiding the `_driver_api` part.